ons-spark/spark-app/spark_application_and_ui.md

```diff
@@ -0,0 +1,429 @@
+|         Cat|                          ASSIST RSPCA WITH CAT STUCK ON CHIMNEY|     295.0|     small|
+|         Dog|                                       DOG FALLEN INTO THE CANAL|     260.0|     small|
+|        Deer|                                          DEER STUCK IN RAILINGS|     520.0|    medium|
+|        Deer|                                           DEER TRAPPED IN FENCE|     260.0|     small|
++------------+----------------------------------------------------------------+----------+----------+
+only showing top 7 rows
+```
+To produce the above `.show()`/`head()` Spark didn't need to process all of the partitions, there are enough rows on one partition to create the output so Spark only processed one partition. If we want Spark to process all of the partitions we need to call an action that involves all rows of the DataFrame, such as `.count()`.
+````{tabs}
+```{code-tab} py
+rescue.count()
+```
+
+```{code-tab} r R
+
+rescue %>% sparklyr::sdf_nrow()
+
+```
+````
+
+```plaintext
+5898
+```
+### Jobs page
+Now that we've created some jobs let's have a quick look at the application's *Event timeline*. For those who are running this as a notebook:
+
+1. Go into the Spark UI
+2. Make sure you're on the Jobs tab
+3. Open the *Event timeline* collapsable section using the blue arrow to see something similar to the below image.
+
+Note the images in this article were created using PySpark, the Spark UI will look slightly different when using sparklyr, but not very different.
+
+In the top section of the timeline you will see the executors being added and removed by the *dynamic allocation* feature of Spark. The `SparkSession` used in this case was a local session, so we will only see the driver here. The jobs are shown in the bottom section of the timeline. Hover over a job in the timeline and the corresponding job in the *Completed Jobs* table below will be highlighted.
+
+*Tip:* You can tick the *Enable Zooming* button to zoom in and out of different sections of the timeline
+
+![Event timeline within Spark UI showing executors being assigned and jobs being deployed](../images/spark_app_event_timeline.png)
+
+Let's look at the *Completed Jobs* table. The description gives us a clue as to the action that initiated that job. The first job, *Job Id* 0, was to interact with HDFS, so it has the description `parquet at NativeMethodAccessorImpl.java:0`. Remember that executing transformations creates an execution plan, so Spark needs to know the DataFrame's schema, i.e. column names and types, to validate our PySpark/sparklyr code. Reading from disk will always create a job, usually consisting of just one stage as shown in the *Stages* column.
+
+The second job was initiated by `.show()`/`head()`, again it consists of one stage, which itself had one task as shown in the *Tasks* column. As mentioned in the previous section, Spark only needed to process one partition to produce the output so we therefore have one task in this job. 
```

Comment:
head() by itself isn't an action; just happens that we're implicitly collecting it to print in this instance. You could change the code to add collect() after head() and say that it was initiated by .show()/collect() possibly?